server/src/main/java/org/elasticsearch/index/codec/PerFieldMapperCodec.java

```diff
@@ -110,33 +110,23 @@ public DocValuesFormat getDocValuesFormatForField(String field) {
     }
 
     boolean useTSDBDocValuesFormat(final String field) {
-        return mapperService.getIndexSettings().isES87TSDBCodecEnabled()
-            && isTimeSeriesModeIndex()
-            && isNotSpecialField(field)
-            && (isCounterOrGaugeMetricType(field) || isTimestampField(field));
-    }
-
-    private boolean isTimeSeriesModeIndex() {
-        return IndexMode.TIME_SERIES.equals(mapperService.getIndexSettings().getMode());
-    }
-
-    private boolean isCounterOrGaugeMetricType(String field) {
-        if (mapperService != null) {
+        if (mapperService != null && mapperService.getIndexSettings().isES87TSDBCodecEnabled() && isTimeSeriesModeIndex()) {
             final MappingLookup mappingLookup = mapperService.mappingLookup();
             if (mappingLookup.getMapper(field) instanceof NumberFieldMapper) {
-                final MappedFieldType fieldType = mappingLookup.getFieldType(field);
-                return TimeSeriesParams.MetricType.COUNTER.equals(fieldType.getMetricType())
-                    || TimeSeriesParams.MetricType.GAUGE.equals(fieldType.getMetricType());
+                return true;
+            }
+            if (mappingLookup.getMapper(field) instanceof DateFieldMapper) {
+                return true;
+            }
+            if (mappingLookup.getMapper(field) instanceof KeywordFieldMapper) {
+                return true;
```

Comment:
Maybe also add:


```
if (mappingLookup.getMapper(field) instanceof TimeSeriesIdFieldMapper) {
   return true;
}
```

I think the _tsid field can benefit a lot from run-length encoding.

Reply:
Good idea. Added.